src/main/java/org/skriptlang/reflect/syntax/CustomSyntaxStructure.java

```diff
@@ -285,20 +182,41 @@ private void update() {
   }
 
   protected final void register(T data) {
+    hasPatterns = true;
     String pattern = data.getPattern();
 
     whichInfo.add(data);
 
     getDataTracker().getPrimaryData()
         .computeIfAbsent(data.getScript(), f -> new HashMap<>())
         .put(pattern, data);
+  }
 
-    update();
+  protected boolean checkHasPatterns() {
+    if (hasPatterns)
+      return true;
+    Skript.error("A custom syntax must have at least one pattern");
+    return false;
+  }
+
+  protected SectionNode[] getParseNode() {
+    SectionNode parseNode = getEntryContainer().getOptional("parse", SectionNode.class, false);
+    SectionNode safeParseNode = getEntryContainer().getOptional("safe parse", SectionNode.class, false);
+    if (parseNode != null && safeParseNode != null) {
+      Skript.error("A custom syntax element cannot contain both 'parse' and 'safe parse' entries");
+      return null;
+    }
+    if (safeParseNode != null) {
+      Skript.warning("The 'safe parse' entry is deprecated and will act as a regular 'parse' entry."
+          + " Please use the 'parse' entry instead");
+      return new SectionNode[] {safeParseNode};
+    }
```

Comment:
```suggestion
    if (afeParseNode != null) {
      if (parseNode != null {
        Skript.error("A custom syntax element cannot contain both 'parse' and 'safe parse' entries");
        return null;
      } else {
        Skript.warning("The 'safe parse' entry is deprecated and will act as a regular 'parse' entry."
          + " Please use the 'parse' entry instead");
        return new SectionNode[] {safeParseNode};
      }
    }
```
maybe this is better